website/docs/connect-data/how-to-guides/fetch-and-filter-data-in-sql.md

```diff
@@ -0,0 +1,255 @@
+
+You can use either a Table or List widget to display the paginated data, and you can implement pagination with offset or cursor-based techniques by using the reference property of these widgets, like:
+
+<dd>
+
+*Examples:* Create a query to fetch data from the database using `pageSize`, and `pageOffset` reference properties to implement pagination:
+
+
+```sql
+SELECT * FROM users LIMIT {{ Table1.pageSize }} OFFSET {{ Table1.pageOffset }};
+```
+
+
+
+</dd>
+
+
+Learn more about [Server-side Pagination on Table](/build-apps/how-to-guides/Server-side-pagination-in-table) and [Server-side Pagination on List](/build-apps/how-to-guides/Setup-Server-side-Pagination-on-List).
+
+## Transform Data
+
+If your retrieved data does not match a widget's expected format, you can either use JavaScript to transform it in the widget property pane or retrieve the required data directly from the query.
+
+<dd>
+
+*Examples:* If you want to display formatted data in a Select widget, where the country name serves as the label and the ID as the value, you can use:
+
+
+```sql
+SELECT country AS name, id AS code
+FROM users
+LIMIT 10;
+```
+
+
+</dd>
+
+
+## Using conditional operators
+
+Conditional operators, like the AND operator, allow you to specify multiple conditions in your SQL queries. This can be handy when you need to narrow down your data retrieval based on several criteria. 
+
+<dd>
+
+*Example:* If you want to apply filters based on country, gender, and a specific date (DOB) using Select and DatePicker widgets.
+
+Configure the query to fetch data using [selectedOptionValue](/reference/widgets/select#selectedoptionvalue-string), [formattedDate](/reference/widgets/datepicker#formatteddate-string) reference properties, like:
+
+```sql
+SELECT * FROM users
+WHERE gender = {{Select1.selectedOptionValue}} AND country = {{Select2.selectedOptionValue}} AND DATE(dob) > {{DatePicker1.formattedDate}};
+```
+
+
+</dd>
+
+
+
+
+
+## Using LIKE or ILIKE 
+
+When using the LIKE or ILIKE keyword in SQL, it's important to handle it correctly, especially when working with prepared statements. 
+
+
+<dd>
+
+
+**Prepared statements enabled** (recommended)
+
+
+```sql
+-- With prepared statements enabled
+SELECT * FROM public.users where name ilike {{Table1.searchText + "%"}}
+```
+
+* The `%` character, which is used as a wildcard in LIKE and ILIKE patterns, needs to be inside the mustache binding `{{}}`.
+* Do not wrap the mustache binding `{{}}` within quotes.
+
+
+</dd>
+
+<dd>
+
+**Prepared statements disabled**
+
+
+
+```sql
+-- With prepared statements disabled
+SELECT * FROM public."features" where name ILIKE '%{{search_input.text}}%' order by created_at desc
+```
+
+
+Placing the `%`character within single quotes can lead to unexpected results and security vulnerabilities. It doesn't provide the same level of security and flexibility as the recommended format, making it more susceptible to SQL injection attacks. 
```

Comment:
I am not feeling confident about this line - `Placing the `%`character within single quotes can lead to unexpected results and security vulnerabilities.` - IMO, the security concern should not be because of where the quotes are placed - it should be only becuase prepared statements are not used - because without prepared statements it becomes vulnerable to SQL injection attacks.